pkgs/development/interpreters/python/tests.nix

```diff
@@ -121,4 +121,38 @@ let
     # in assert myPackages.foobar == myPackages.numpy; myPackages.python.withPackages(ps: with ps; [ foobar ]);
   };
 
-in lib.optionalAttrs (stdenv.hostPlatform == stdenv.buildPlatform ) (environmentTests // integrationTests // overrideTests)
+  condaTests = let
+    requests = callPackage ({
+        autoPatchelfHook,
+        fetchurl,
+        python38,
```

Comment:
The tests are executed for each Python interpreter. Therefore, here you should pass `python` in, and then disable the test for Python versions not supported. See a bit above where Python versions are limited as well.

Reply:
Thanks. I now modified the test to work for all interpreter versions.